crates/bevy_solari/src/pathtracer/pathtracer.wgsl

```diff
@@ -37,30 +37,42 @@ fn pathtrace(@builtin(global_invocation_id) global_id: vec3<u32>) {
     // Path trace
     var radiance = vec3(0.0);
     var throughput = vec3(1.0);
+    var p_bounce = 0.0;
+    var is_perfectly_specular = true;
+    var previous_normal = vec3(0.0);
     loop {
         let ray_hit = trace_ray(ray_origin, ray_direction, ray_t_min, RAY_T_MAX, RAY_FLAG_NONE);
         if ray_hit.kind != RAY_QUERY_INTERSECTION_NONE {
             let ray_hit = resolve_ray_hit_full(ray_hit);
             let wo = -ray_direction;
 
-            // Use emissive only on the first ray (coming from the camera)
-            if ray_t_min == 0.0 { radiance = ray_hit.material.emissive; }
+            var mis_weight = 1.0;
+            if !is_perfectly_specular {
```

Comment:
Why does the bounce being a perfectly specular reflection matter?

Reply:
BSDF PDF sampling will then return 0.0 as the pdf, which, if p_light is small, would lead to fireflies or inf/nans.

Reply:
In general, we wanna disable MIS for dirac delta functions, so it might be a good idea to special-case the BRDF lobe MIS for perfect specular reflections too.

Reply:
I think you might want to modify the mix function so that 0 roughness = always selects specular.

Reply:
Can't do that, we still need to select diffuse if metallic is not 1, I just added a special case if statement at the top.